src/evaluateXPath.ts

```diff
@@ -124,13 +128,64 @@ function transformXPathItemToJavascriptObject(value, dynamicContext) {
 }
 
 export type Options = {
-	disableCache?: boolean;
+	debug?: boolean;
 	language?: Language;
 	moduleImports?: { [s: string]: string };
 	namespaceResolver?: (s: string) => string | null;
 	nodesFactory?: INodesFactory;
 };
 
+function printAndRethrowError(selector: string, error: Error): never {
+	if (error instanceof Error) {
+		throw new Error(`Error executing XPath: ${error}`);
+	}
+	// This must be a StackTraceEntry 'error'
+	const stackEntry = error as StackTraceEntry;
+
+	const errorLocation = stackEntry.getErrorLocation();
+	const rawLines = selector.replace('\r', '').split('\n');
+	const lines = rawLines.reduce((markedLines: string[], line, i) => {
+		const lineNumber = i + 1;
+
+		// Only mark 2 lines before and after
+		if (errorLocation.start.line - lineNumber > 2 || lineNumber - errorLocation.end.line > 2) {
+			return markedLines;
+		}
+
+		const prefix = `${Array(rawLines.length.toString().length)
```

Comment:
We could also get the width of number (prefix) using logarithms: `const linePrefixWidth = Math.floor(Math.log10(Math.min(errorLocation.end.line + 2, rawLines.length))) + 1`. This may be a bit more readable.